plasmapy/particles/particle_class.py

```diff
@@ -1815,6 +1892,41 @@ def __repr__(self):
         """
         return f"CustomParticle(mass={self.mass}, charge={self.charge})"
 
+    def to_dict(self) -> dict:
+        """
+        Return a dictionary representation of the minimum description of a particle.
+
+        Examples
+        --------
+        >>> custom_particle = CustomParticle(mass=1.5e-26 * u.kg, charge=-1)
+        >>> custom_particle.to_dict()
+        {'type': 'CustomParticle', 'mass': {'quantity': 1.5e-26, 'unit': 'kg'}, 'charge': {'quantity': -1.602176634e-19, 'unit': 'C'}}
+        >>> custom_particle = CustomParticle(mass=1.5e-26 * u.kg)
+        >>> custom_particle.to_dict()
+        {'type': 'CustomParticle', 'mass': {'quantity': 1.5e-26, 'unit': 'kg'}, 'charge': {'quantity': nan, 'unit': 'C'}}
+
+        """
+        particle_dict = dict()
+        particle_dict['type'] = 'CustomParticle'
+        particle_dict['mass'] = {"quantity": self.mass.value, "unit": str(self.mass.unit)}
+        particle_dict['charge'] = {"quantity": self.charge.value, "unit": str(self.charge.unit)}
```

Comment:
Looks like this could be a bit simpler:

```python
[ins] In [1]: u                                                                                                                           
Out[1]: <module 'astropy.units' from '/usr/lib/python3.8/site-packages/astropy/units/__init__.py'>

[ins] In [2]: 2 * u.kg                                                                                                                    
Out[2]: <Quantity 2. kg>

[ins] In [3]: mass = 2 * u.kg                                                                                                             

[ins] In [4]: str(mass)                                                                                                                   
Out[4]: '2.0 kg'

[ins] In [5]: u.Quantity(str(mass))                                                                                                       
Out[5]: <Quantity 2. kg>

[ins] In [6]: _.unit                                                                                                                      
Out[6]: Unit("kg")

[ins] In [7]: __.value                                                                                                                    
Out[7]: 2.0
```

But I suppose it's fine the way it is, as well! As long as we're consistent. I just want to make sure that we don't end up introducing custom conventions separate from Astropy's.

Reply:
> ```python
> [ins] In [5]: u.Quantity(str(mass))                                                                                                      
> Out[5]: <Quantity 2. kg> ```

Thank you for suggesting this. I was under the impression that we'd need to manually reform the quantity by doing `magnitude * u.Quantity`. The method you have suggested is definitely easier to code as well as maintain in future, so I shall adopt that into my code.

As an added bonus, this is also likely to remove the pep8 warnings about the extra long lines xD.